File: codewars/kyu_6/python/find_the_parity_outlier.py
# https://www.codewars.com/kata/find-the-parity-outlier/

def find_outlier(integers):
    signs = [bool(x % 2) for x in integers[:3]]
    entirely_odd = True if signs.count(True) > signs.count(False) else False
    
    for x in integers:
        if entirely_odd and not x % 2:
            return x
        if not entirely_odd and x % 2:
            return x

File: codewars/kyu_6/php/find_the_parity_outlier.php
<?php

# https://www.codewars.com/kata/find-the-parity-outlier/

function find($integers) {
    $odds = array();
    $evens = array();
    foreach ($integers as $elem) {
        if ($elem % 2) {
            array_push($odds, $elem);
        }
        else {
            array_push($evens, $elem);
        }
    }
    return sizeof($odds) < sizeof($evens) ? $odds[0] : $evens[0];
}

?>
